processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/ProcessingContextImpl.java

```diff
@@ -92,16 +91,23 @@ public String subscriptionId() {
     }
 
     @Override
-    public DeferredCompletion deferCompletion() {
+    public DeferredCompletion deferCompletion(Consumer<ProcessingContext<T>> timeoutCallback) {
```

Comment:
Hm, I wonder if this callback signature is confusing, because there are two `ProcessingContext`s inside the callback like below, so it's not clear which is appropriate. :
```java
void process(ProcessingContext context, Task task) {
    context.deferComplection(context2 -> {
        // there are context and context2
    });
}
```

Though in current implementation, they refers same `ProcessingContext`.
Just `Runnable` would be simple?

Reply:
Hm, maybe yes. Let me think.